libp2p/network/swarm.py

```diff
@@ -87,17 +88,53 @@ def set_stream_handler(self, stream_handler: StreamHandlerFn) -> None:
         try:
             # Get peer info from peer store
             addrs = self.peerstore.addrs(peer_id)
-        except PeerStoreError:
-            raise SwarmException(f"No known addresses to peer {peer_id}")
+        except PeerStoreError as error:
+            raise SwarmException(f"No known addresses to peer {peer_id}") from error
 
-        if not addrs:
-            raise SwarmException(f"No known addresses to peer {peer_id}")
+        exceptions: List[SwarmException] = []
+
+        # Try all known addresses
+        for multiaddr in addrs:
+            try:
+                return await self.dial_addr(multiaddr, peer_id)
+            except SwarmException as e:
+                exceptions.append(e)
+                logger.debug(
+                    "encountered swarm exception when trying to connect to %s, "
+                    "trying next address...",
+                    multiaddr,
+                    exc_info=e,
+                )
+
+        # Tried all addresses, raising exception.
+        if len(exceptions) > 0:
+            raise SwarmException(
+                "unable to connect to %s, no addresses established a successful connection "
```

Comment:
Is this a typo? I think `SwarmException` is not able to resolve the format string. Besides, we favor f-string and `.format` over `%s` style format string.

Reply:
My thought was that you guys used `%s` for exception and log formatting, I'll correct this.